core/eolearn/core/eodata_io.py

```diff
@@ -154,6 +158,45 @@ def walk_eopatch(eopatch, patch_location, features=...):
         else:
             yield ftype, fname, fs.path.combine(name_basis, fname)
 
+def merge_eopatch(eopatch_features, filesystem_features, eopatch, fs_eopatch):
+    """ Concatenate an existing eopatch in the filesystem and a new eopatch chronologically
+    """
+
+    filesystem_features = {_to_lowercase(*feature) for feature in filesystem_features}
+    eopatch_features = {_to_lowercase(*feature) for feature in eopatch_features}
+
+    intersection = filesystem_features.intersection(eopatch_features)
+    if intersection:
+        time_periods = [(fs_eopatch.timestamp[0].strftime("%Y-%m-%d"), fs_eopatch.timestamp[-1].strftime("%Y-%m-%d")),
+                        (eopatch.timestamp[0].strftime("%Y-%m-%d"), eopatch.timestamp[-1].strftime("%Y-%m-%d"))]
+
+        mask = np.isin(fs_eopatch.timestamp, list(set(fs_eopatch.timestamp).difference(set(eopatch.timestamp))))
+
+        timestamp = eopatch.timestamp + [tstamp for tstamp, to_keep in zip(fs_eopatch.timestamp, mask) if to_keep]
+        sorted_indices = sorted(range(len(timestamp)), key=lambda k: timestamp[k])
+
+        eopatch.timestamp = sorted(timestamp)
+
+        for ftype, _ in intersection:
+            if ftype in [FeatureType.DATA, FeatureType.MASK, FeatureType.SCALAR]:
+                for fname in eopatch[ftype]:
+                    feat = (ftype, fname)
+                    if eopatch[feat].shape[1:] == fs_eopatch[feat].shape[1:]:
+                        array = np.concatenate((eopatch[feat], fs_eopatch[feat][mask]), axis=0)
+                        eopatch[feat] = array[sorted_indices]
+            elif ftype in [FeatureType.DATA_TIMELESS, FeatureType.MASK_TIMELESS, FeatureType.SCALAR_TIMELESS]:
+                for fname in eopatch[ftype]:
+                    feat = (ftype, fname)
+                    if eopatch[feat].shape == fs_eopatch[feat].shape:
+                        eopatch[feat] = np.nanmean((eopatch[feat], fs_eopatch[feat]), axis=0)\
+                            .astype(eopatch[feat].dtype)
```

Comment:
Should there perhaps be some warning printed out which notifies the user which features (if any) were ignored? Do you think this makes sense?